corehq/sql_db/connections.py

```diff
@@ -151,26 +170,20 @@ def _populate_connection_map(self):
                 if isinstance(db_config, dict):
                     write_db = db_config['WRITE']
                     read = db_config['READ']
-                    for db_alias, weighting in read:
-                        assert isinstance(weighting, int), 'weighting must be int'
-                        assert db_alias in settings.DATABASES, db_alias
+                    assert set(read.keys()).issubset(set(settings.DATABASES.keys()))
 
                 self._add_django_db(engine_id, write_db)
                 if read:
-                    self.read_database_mapping[engine_id] = []
+                    self.read_database_mapping[engine_id] = read
                     for read_db, weighting in read:
                         assert read_db == write_db or read_db not in self.db_connection_map, read_db
-                        self.read_database_mapping[engine_id].extend([read_db] * weighting)
                         if read_db != write_db:
                             self._add_django_db(read_db, read_db)
 
         for app, weights in settings.LOAD_BALANCED_APPS.items():
-            self.read_database_mapping[app] = []
-            for db_alias, weighting in weights:
-                assert isinstance(weighting, int), 'weighting must be int'
-                assert db_alias in settings.DATABASES, db_alias
-
-                self.read_database_mapping[app].extend([db_alias] * weighting)
+            self.read_database_mapping[app] = weights
+            dbs = weights.keys()
+            assert set(dbs).issubset(set(settings.DATABASES.keys()))
```

Comment:
same here re `.keys()` and the `int` check

Reply:
I will update.